contrib/cmake/build3rdParty/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+
+
+cmake_minimum_required(VERSION 3.13)
+project(Build3rdParty)
+
+if (WIN32)
+    # accurate __cplusplus macro for vc++, selecting c++17 here for windows builds though the MEGA SDK library must build for older c++ standards also
```

Comment:
We should consider using the `target_meow` approach here, e.g.,

```cmake
target_compile_features(build3rdParty cxx_std_11 REQUIRED)
target_compile_features(build3rdParty cxx_std_17 REQUIRED)
target_compile_options(build3dParty /Zc:__cplusplus)
```

Also I haven't developed on MSVC in a bit over a year--is support for `/Zc:__cplusplus` more widespread now? There used to be some libraries that detected if they were running on MSVC by checking for `__cplusplus` being 1997

Reply:
I think this is the sort of thing we would adjust if any problem emerged - for windows we're using VS 2017/2019 and both seem fine.